lib/mongoid/clients/options.rb

```diff
@@ -44,12 +44,12 @@ def persistence_options
 
       def mongo_client
         tmp = persistence_options
-        if opts = tmp && tmp.dup
+        if (opts = tmp && !tmp.empty? && tmp.dup)
           if opts[:client]
             client = Clients.with_name(opts[:client])
           else
             client = Clients.with_name(self.class.client_name)
-            client.use(self.class.database_name)
+            client = client.use(self.class.database_name)
```

Comment:
```
def use(name)
  with(database: name)
end
```

Unlike `moped`, `mongo` does not switch database by calling `use`, you need to use the new client.